xmtp_id/src/scw_verifier/chain_rpc_verifier.rs

```diff
@@ -87,14 +87,20 @@ impl SmartContractSignatureVerifier for RpcSmartContractWalletVerifier {
         ];
         let data = constructor.encode_input(code, tokens)?;
         let tx: TypedTransaction = TransactionRequest::new().data(data).into();
-        let block_number = if let Some(number) = block_number {
-            number
-        } else {
-            BlockNumber::Number(self.provider.get_block_number().await?)
+        let block_number = match block_number {
+            Some(bn) => bn,
+            None => BlockNumber::Number(self.current_block_number(&signer).await?),
         };
         let res = self.provider.call(&tx, Some(block_number.into())).await?;
         Ok(res == Bytes::from_hex("0x01").expect("Hardcoded hex will not fail."))
     }
+
+    async fn current_block_number(&self, _account_id: &AccountId) -> Result<U64, VerifierError> {
```

Comment:
I like this approach, but we shouldn't need the account_id to get a block_number


The block number is independent of any account, although it will differ based on `chain_rpc_url` so I think the rest is correct but we can remove this part

Reply:
That's how I had it initially, but saw that the account_id is being used to get the appropriate `verifier` [here](https://github.com/xmtp/libxmtp/pull/1112/files#diff-696d43c0087a23db3e463f659837d15a5fedf997f09aaedb9aaa612b9ac5b697R139)